download-maven-plugin/src/main/java/com/googlecode/WGet.java

```diff
@@ -299,8 +315,19 @@ private void doGet(File outputFile) throws Exception {
     // TODO: this should be retrieved from wagonManager
     com.googlecode.ConsoleDownloadMonitor downloadMonitor = new com.googlecode.ConsoleDownloadMonitor();
     wagon.addTransferListener(downloadMonitor);
-    wagon.connect(repository,
-        this.wagonManager.getProxy(repository.getProtocol()));
+    if (StringUtils.isNotBlank(username)) {
+      getLog().info("providing custom authentication");
+      getLog().debug("username: " + username + " and password: ***");
+      AuthenticationInfo authInfo = new AuthenticationInfo();
+      authInfo.setUserName(username);
+      authInfo.setPassword(password);
```

Comment:
If setting a passwork mandatory at that time? Ie can we have some AuthentificationInfo with only a userName and no password?
Is there a difference between no password, a null password and a blank password?